gateway/mw_go_plugin.go

```diff
@@ -255,9 +255,18 @@ func (m *GoPluginMiddleware) goPluginFromTykVersion(version string) string {
 	if len(vs) > 0 {
 		version = vs[0]
 	}
+	version = EnsureSemanticVersioning(version)
 
 	newPluginName := strings.Join([]string{pluginName, version, os, architecture}, "_")
 	newPluginPath := pluginDir + "/" + newPluginName + ".so"
 
 	return newPluginPath
 }
+
+// EnsureSemanticVersioning receives a version and check that it has the prefix 'v' otherwise, it adds it
+func EnsureSemanticVersioning(version string) string {
+	if !strings.HasPrefix(version, "v") {
```

Comment:
Hey @sredxny, what do you think about adding a second conditional to check if the letter "v" is in the first place of the string? I mean, the version could be "1.3.3v" and the function wouldn't detect it. Btw, I don't know if that's something possible to happen, just asking :)

Reply:
@mativm02 that is exactly what `strings.HasPrefix` does. It checks that the string starts with "v"

Reply:
@sredxny I don’t know why I thought you were using strings.Contains(). Sorry for the confusion.